lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -1875,7 +1828,9 @@ Scope * ByteCodeGenerator::FindScopeForSym(Scope *symScope, Scope *scope, Js::Pr
 {
     for (scope = scope ? scope->GetEnclosingScope() : currentScope; scope; scope = scope->GetEnclosingScope())
     {
-        if (scope->GetFunc() != funcInfo && scope->GetMustInstantiate() && scope != this->globalScope)
+        if (scope->GetFunc() != funcInfo
+            && (scope->GetMustInstantiate() || (scope->GetScopeType() == ScopeType_Parameter && !scope->GetCanMergeWithBodyScope()))
```

Comment:
Ah. We have 2 choices here: have the body and param scope both in the scope stack, and both counted; or have the param scope popped off the stack before we begin emitting the body of the function. We settled on the first one for ease of implementation, right? In that case, the code that accesses outer scopes should count them both. Making the param scope a special case in this calculation is asking for trouble.